### PR TITLE
move everything not explicitly riot (or status) branded into matrix-react-sdk

### PR DESCRIPTION
**Pull request by ara4n**
This moves almost everything not explicitly riot (or status) branded into matrix-react-sdk, in order to fix the layering of riot-web v. matrix-react-sdk, as per https://docs.google.com/document/d/1itOIy6zdKGJCQZEPecL9uTPTtBCVrZ_HgwQiDnah_LM.

For now we're deliberately keeping the `vector/` directory (i.e. the top level app skin itself) in the riot-web repo.  In future we might want to provide a simple sample skin app in matrix-react-sdk.

Status:
 * [x] this moves the files into the right(?) place
 * [x] actually make it build again.
 * [x] fix the tests
 * [x] remerge the i18n

**Whilst this PR is in flight, I suggest folks try to avoid making changes to the riot-web repository.**

Twinned with https://github.com/matrix-org/matrix-react-sdk/pull/1836

### CODE REVIEW

**Comment by martindale** [comment]
Recommend: prefix in-flight PRs with `WIP: ` in their title.  This looks awesome, otherwise!

File: test/unit-tests/README.md [hunk 1]
```diff
@@ -0,0 +1,2 @@
+Any UTs for vector-web layer components or functionality should go here.
+This used to contain the UTs for notifications before they got moved to react-sdk.
```
Comment (lukebarnard1) [inline test/unit-tests/README.md]: Commit message is misleading here. Should there be changes to the tests as part of this commit?

File: src/components/structures/VectorHomePage.js [hunk 2]
```diff
@@ -24,7 +24,10 @@ import sanitizeHtml from 'sanitize-html';
 import sdk from 'matrix-react-sdk/lib';
 
 module.exports = React.createClass({
-    displayName: 'HomePage',
+    displayName: 'VectorHomePage',
+    statics: {
+        replaces: 'HomePage',
```
Comment (lukebarnard1) [inline src/components/structures/VectorHomePage.js]: This looks like it needs a comment.

File: test/unit-tests/README.md (hunk 1 above)
Comment (ara4n) [reply test/unit-tests/README.md]: sorry, this was a terrible commit message. the thing that was breaking the tests is that the `unit-tests` dir had winked out of existence. creating this README file recreated the directory and fixed the tests O:-)

File: src/components/structures/VectorHomePage.js (hunk 2 above)
Comment (ara4n) [reply src/components/structures/VectorHomePage.js]: i forgot about this. I need to turn HomePage into a ES6 class and then have VectorHomePage inherit from it (whilst also `replace`ing it)

File: src/components/structures/VectorHomePage.js [hunk 3]
```diff
@@ -18,97 +18,22 @@ limitations under the License.
...
+import { _t } from 'matrix-react-sdk/lib/languageHandler';
 
-    getInitialState: function() {
-        return {
-            iframeSrc: '',
-            page: '',
-        };
-    },
+class VectorHomePage extends HomePage {
+    displayName = 'VectorHomePage'
```
Comment (dbkr) [inline src/components/structures/VectorHomePage.js]: Should this have a 'static' too? Also I think these wants semicolons.

File: src/vector/index.js [hunk 4]
```diff
@@ -32,10 +32,10 @@ require('babel-polyfill');
...
 // Our own CSS (which is themed) is imported via separate webpack entry points
 // in webpack.config.js
-require('gemini-scrollbar/gemini-scrollbar.css');
-require('gfm.css/gfm.css');
-require('highlight.js/styles/github.css');
-require('draft-js/dist/Draft.css');
+require('matrix-react-sdk/node_modules/gemini-scrollbar/gemini-scrollbar.css');
+require('matrix-react-sdk/node_modules/gfm.css/gfm.css');
+require('matrix-react-sdk/node_modules/highlight.js/styles/github.css');
+require('matrix-react-sdk/node_modules/draft-js/dist/Draft.css');
```
Comment (dbkr) [inline src/vector/index.js]: I assume this didn't work without hardcoding the path? I think this might break between versions of npm.

Comment (dbkr) [reply src/vector/index.js]: oh yep, I thought I'd seen that somewhere else. I don't know off hand any other way to do this, I guess other than importing the dependency directly.

**Pull request merged**